internals/secrethub/service_aws_init.go

```diff
@@ -0,0 +1,95 @@
+package secrethub
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+
+	"github.com/secrethub/secrethub-cli/internals/cli/ui"
+
+	"github.com/secrethub/secrethub-go/internals/api"
+	"github.com/secrethub/secrethub-go/internals/errio"
+)
+
+// Errors
+var (
+	ErrInvalidAWSRegion = errMain.Code("invalid_region").Error("invalid AWS region")
+	ErrRoleAlreadyTaken = errMain.Code("role_taken").Error("a service using that IAM role already exists")
+)
+
+// ServiceAWSInitCommand initializes a service for AWS.
+type ServiceAWSInitCommand struct {
+	description string
+	path        api.DirPath
+	kmsKeyID    string
+	role        string
+	region      string
+	permission  api.Permission
+	io          ui.IO
+	newClient   newClientFunc
+}
+
+// NewServiceAWSInitCommand creates a new ServiceAWSInitCommand.
+func NewServiceAWSInitCommand(io ui.IO, newClient newClientFunc) *ServiceAWSInitCommand {
+	return &ServiceAWSInitCommand{
+		io:        io,
+		newClient: newClient,
+	}
+}
+
+// Run initializes an AWS service.
+func (cmd *ServiceAWSInitCommand) Run() error {
+	repo := cmd.path.GetRepoPath()
+
+	client, err := cmd.newClient()
+	if err != nil {
+		return err
+	}
+
+	cfg := aws.NewConfig()
+	if cmd.region != "" {
+		_, ok := endpoints.AwsPartition().Regions()[cmd.region]
+		if !ok {
+			return ErrInvalidAWSRegion
+		}
+		cfg = cfg.WithRegion(cmd.region)
+	}
+
+	service, err := client.Services().AWS().Create(repo.Value(), cmd.description, cmd.kmsKeyID, cmd.role, cfg)
+	if err == api.ErrCredentialAlreadyExists {
+		return ErrRoleAlreadyTaken
+	} else if err != nil {
+		return err
+	}
+
+	if cmd.permission != 0 {
+		_, err = client.AccessRules().Set(cmd.path.Value(), cmd.permission.String(), service.ServiceID)
+		if err != nil {
+			_, delErr := client.Services().Delete(service.ServiceID)
+			if delErr != nil {
+				fmt.Fprintf(cmd.io.Stdout(), "Failed to cleanup after creating an access rule for %s failed. Be sure to manually remove the created service account %s: %s\n", service.ServiceID, service.ServiceID, err)
+				return errio.Error(delErr)
+			}
+
+			return errio.Error(err)
+		}
+	}
+
+	fmt.Fprintf(cmd.io.Stdout(), "The service %s is now reachable through AWS when the role %s is assumed.\n", service.ServiceID, cmd.role)
```

Comment:
Consider something like: `Successfully created a service account. Any host that assumes the IAM role %s can now automatically authenticate to SecretHub and fetch the secrets it has been given access to.`

Reply:
:+1: 

I've replaced the last `it` with `the service`, as I read it as referring to the host.

`Successfully created a service account. Any host that assumes the IAM role %s can now automatically authenticate to SecretHub and fetch the secrets the service has been given access to.`